File: RNCalculator/ios/CalculatorModule.swift
//
//  CalculatorModule.swift
//  RNCalculator
//
//  Created by delay on 2023/03/02.
//

import Foundation

@objc(CalculatorModule)
class CalculatorModule:NSObject {
  
  
  @objc(executeCalc:numberA:numberB:resolver:rejector:)
  public func executeCalc(_ action:String, numberA:Int, numberB:Int, resolver:RCTPromiseResolveBlock, rejector:RCTPromiseRejectBlock) -> Void{
    
    
    if(action == "plus"){
      resolver(numberA + numberB);
      return;
    }
    
    if(action == "minus"){
      resolver(numberA - numberB);
      return;
    }
    
    if(action == "multiply"){
      resolver(numberA * numberB);
      return;
    }
    
    if(action == "divide"){
      resolver(numberA / numberB);
      return;
    }
    
    rejector("Unexpected action type", action, nil);
    
    
  }
}

File: RNCalculator/android/app/src/main/java/com/rncalculator/CalculatorModule.kt
package com.rncalculator

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class CalculatorModule(reactContext:ReactApplicationContext):ReactContextBaseJavaModule(reactContext) {
    override fun getName() = "CalculatorModule"

    @ReactMethod
    fun executeCalc(action:String, numberA:Double, numberB:Double, promise: Promise){
        if(action == "plus"){
            promise.resolve(numberA + numberB);
            return;
        }

        if(action == "minus"){
            promise.resolve(numberA - numberB);
            return;
        }

        if(action == "multiply"){
            promise.resolve(numberA * numberB);
            return;
        }

        if(action == "devide"){
            promise.resolve(numberA/numberB);
            return;
        }

        promise.reject("Unexpected action type")
    }
}